lib/module.js

```diff
@@ -74,6 +74,15 @@ function Module(id, parent) {
 }
 module.exports = Module;
 
+const builtinModules = Object.keys(process.binding('natives'))
+  .filter((module) => {
+    return !module.startsWith('internal/') &&
+      !module.startsWith('v8/') &&
+      !module.startsWith('node-inspect/');
+  });
+
```

Comment:
How about the underscore'd (such as `_tls_legacy`)modules? should we inhibit them from being exposed?
This is the output with the current code:
```
[ 'async_hooks',
  'assert',
  'buffer',
  'child_process',
  'console',
  'constants',
  'crypto',
  'cluster',
  'dgram',
  'dns',
  'domain',
  'events',
  'fs',
  'http',
  '_http_agent',
  '_http_client',
  '_http_common',
  '_http_incoming',
  '_http_outgoing',
  '_http_server',
  'https',
  'inspector',
  'module',
  'net',
  'os',
  'path',
  'process',
  'punycode',
  'querystring',
  'readline',
  'repl',
  'stream',
  '_stream_readable',
  '_stream_writable',
  '_stream_duplex',
  '_stream_transform',
  '_stream_passthrough',
  '_stream_wrap',
  'string_decoder',
  'sys',
  'timers',
  'tls',
  '_tls_common',
  '_tls_legacy',
  '_tls_wrap',
  'tty',
  'url',
  'util',
  'v8',
  'vm',
  'zlib' ]
```

Reply:
Personally, +1 for exposing them, since they are files that can be required, and we wouldn't want  a module name to conflict.

Reply:
By the way, there is also `repl._builtinLibs` which I guess lives in the somewhat-documented-but-not-really-officially-supported zone, which only has the modules defined in the REPL/for `--eval` … I think that could cover the use case of “these are the libraries you actually want to use”?

Reply:
i think it would make more sense to expose internal module's builtin lib array